plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -596,6 +623,22 @@ public LocatedTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
                 tableSnapshot.getVersion());
     }
 
+    private void cacheTableMetadataAsync(Table table)
+    {
+        if (!cacheTableMetadata) {
+            return;
+        }
+
+        cacheTableMetadataExecutor.execute(() -> {
```

Comment:
If the table we're reading has the same version as in the metastore, there's no reason to "replace" it.

Think of the scenario:

```
SELECT * FROM table1;
SELECT * FROM table1;
SELECT * FROM table1;
SELECT * FROM table1;
```
Only for the first query (if there is a disparity between the metastore Table `TRINO_LAST_TRANSACTION_VERSION` property and the `DeltaLakeTableHandle#readVersion` should we actually proceed to cache the metadata.